### PR TITLE
feat: separate auction model and eip1559

### PR DESCRIPTION
**Pull request by koonopek**

### CODE REVIEW

File: packages/rpc-providers/src/TransactionDeliveryMan.ts [hunk 1]
```diff
@@ -61,16 +67,18 @@ type TransactionDeliveryManOpts = {
    */
   percentileOfPriorityFee?: number;
 
+  isAuctionModel?: boolean;
+
   logger?: (text: string) => void;
 };
 
-type FeeStructure = {
+export type Eip1559Fee = {
```
Comment (Lukasz2891) [inline packages/rpc-providers/src/TransactionDeliveryMan.ts:75]: We have Eip1559Fee defined here, but AuctionModelFee is defined in AuctionModelGasEstimator file ;)

File: packages/rpc-providers/src/TransactionDeliveryMan.ts [hunk 2]
```diff
@@ -61,16 +67,18 @@ type TransactionDeliveryManOpts = {
    */
   percentileOfPriorityFee?: number;
 
+  isAuctionModel?: boolean;
```
Comment (Lukasz2891) [inline packages/rpc-providers/src/TransactionDeliveryMan.ts:70]: Maybe an enum?

File: packages/rpc-providers/src/TransactionDeliveryMan.ts [hunk 3]
```diff
@@ -2,24 +2,30 @@ import { ErrorCode } from "@ethersproject/logger";
...
   nonce: number;
   maxFeePerGas: number;
   result?: TransactionResponse;
 };
 
-type FeeHistoryResponse = { reward: string[] };
+export type FeeHistoryResponse = { reward: string[] };
 
-type GasOracleFn = (opts: TransactionDeliveryManOpts) => Promise<FeeStructure>;
+type GasOracleFn = (opts: TransactionDeliveryManOpts) => Promise<Eip1559Fee>;
```
Comment (Lukasz2891) [inline packages/rpc-providers/src/TransactionDeliveryMan.ts:26]: Maybe to restore (renamed) `FeeStructure` and define it as `Eip1559Fee | AuctionModelFee`

File: packages/rpc-providers/src/GasEstimator.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { providers } from "ethers";
+
+export interface GasEstimator<T> {
```
Comment (Lukasz2891) [inline packages/rpc-providers/src/GasEstimator.ts:3]: Don't we expect here to have `T` extending `FeeStructure`?

File: packages/rpc-providers/src/TransactionDeliveryMan.ts [hunk 5]
```diff
@@ -201,17 +216,17 @@ export class TransactionDeliveryMan {
 
   private async getFees(
     provider: providers.JsonRpcProvider
-  ): Promise<FeeStructure> {
+  ): Promise<Eip1559Fee> {
```
Comment (Lukasz2891) [inline packages/rpc-providers/src/TransactionDeliveryMan.ts:219]: Is it important for `TransactionDeliveryMan` to specify the type here or can be used any `FeeStructure` (after having it restored ;p)?

**Comment by Lukasz2891** [comment]
What do u think about creating unit tests to show the differences :)?

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 2 above)
Comment (koonopek) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:70]: Nit pick: why?

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 1 above)
Comment (koonopek) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:75]: Good catch

File: packages/rpc-providers/src/GasEstimator.ts (hunk 4 above)
Comment (koonopek) [reply packages/rpc-providers/src/GasEstimator.ts:3]: good point!

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 5 above)
Comment (koonopek) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:219]: fixed

**Review by cehali — APPROVED** [review]

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 2 above)
Comment (Lukasz2891) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:72]: extendability ;) 

maybe - not so important for me ;p

**Review by Lukasz2891 — APPROVED** [review]

**Pull request merged**